src/core/sources/RTMP.ts

```diff
@@ -304,7 +310,127 @@ export const RTMP = {
           engineSocket.connect()
         }
       } else {
-        // TODO: handle WebRTC preview
+      }
+    })
+
+    CoreContext.on('RoomJoined', ({ projectId, room }) => {
+      const project = toBaseProject(getProject(projectId))
+      if (project.role !== SDK.Role.ROLE_RENDERER) {
+        let previousParticipants = [] as SDK.Participant[]
+        /*  Updating the source of the camera. 
+            Adding multi camera support
+         */
+        const updatePreviewStreams = () => {
+          previousTracks
+            .filter((p) => p?.type === 'screen_share' && p?.isExternal === true)
+            .forEach((track) => {
+              if (track.type === 'screen_share') {
+                const srcObject = rtmpSourceStreams[track.participantId]
+                const participant = room.getParticipant(track.participantId)
+                // only do anything if it is for an RTMP source
+                if (previousRTMPSources.some((source) => source.id === participant.id)) {
+                  const videoTrack = room.getTrack(track.id)
+                  const source = getSource(`rtmp-${participant?.id}`)
+                  if (source) {
+                    const audioTrack = room.getTrack(
```

Comment:
why we are reading the microphone track from the participant meta?

Reply:
Good point. Fixed this.